hazelcast/src/main/java/com/hazelcast/internal/diagnostics/StoreLatencyPlugin.java

```diff
@@ -251,31 +247,26 @@ private LatencyProbeImpl(String methodName, InstanceProbes instanceProbes) {
 
         @Override
         public void recordValue(long durationNanos) {
-            stats.recordValue(durationNanos);
+            distribution.recordNanos(durationNanos);
         }
 
         private void render(DiagnosticsLogWriter writer) {
-            Statistics stats = this.stats;
-            long invocations = stats.count;
-            long totalMicros = stats.totalMicros;
-            long avgMicros = invocations == 0 ? 0 : totalMicros / invocations;
-            long maxMicros = stats.maxMicros;
-
-            if (invocations == 0) {
+            LatencyDistribution stats = this.distribution;
+            if (stats.count() == 0) {
                 return;
             }
 
             writer.startSection(methodName);
-            writer.writeKeyValueEntry("count", invocations);
-            writer.writeKeyValueEntry("totalTime(us)", totalMicros);
-            writer.writeKeyValueEntry("avg(us)", avgMicros);
-            writer.writeKeyValueEntry("max(us)", maxMicros);
+            writer.writeKeyValueEntry("count", stats.count());
+            writer.writeKeyValueEntry("totalTime(us)", stats.totalMicros());
+            writer.writeKeyValueEntry("avg(us)", distribution.avgMicros());
+            writer.writeKeyValueEntry("max(us)", stats.maxMicros());
 
             writer.startSection("latency-distribution");
-            for (int k = 0; k < stats.latencyDistribution.length(); k++) {
-                long value = stats.latencyDistribution.get(k);
+            for (int bucketIndex = 0; bucketIndex < stats.bucketCount(); bucketIndex++) {
+                long value = stats.bucket(bucketIndex);
                 if (value > 0) {
-                    writer.writeKeyValueEntry(LATENCY_KEYS[k], value);
+                    writer.writeKeyValueEntry(LATENCY_KEYS[bucketIndex], value);
```

Comment:
You can use `LatencyDistribution.LATENCY_KEYS[bucket]` and then you can remove the whole `LATENCY_KEYS` field here in this class as it's otherwise not used.

Reply:
Fixed